hack/make-rules/helpers/cache_go_dirs.sh

```diff
@@ -28,7 +28,7 @@ if [[ -z "${1:-}" ]]; then
 fi
 CACHE="$1"; shift
 
-trap "rm -f '${CACHE}'" HUP INT TERM ERR
+trap 'rm -f "${CACHE}"' HUP INT TERM ERR
```

Comment:
Why did it change and what did it fix:
```
In ./hack/make-rules/helpers/cache_go_dirs.sh line 31:
trap "rm -f '${CACHE}'" HUP INT TERM ERR
             ^------^ SC2064: Use single quotes, otherwise this expands now rather than when signalled.
Ref: https://github.com/koalaman/shellcheck/wiki/SC2064
```
Example
**Problematic code**:
```
trap "echo \"Finished on $(date)\"" EXIT
```
**Correct code:**:
```
trap 'echo "Finished on $(date)"' EXIT
```